application.go

```diff
@@ -38,9 +38,9 @@ type Application interface {
 }
 
 type application struct {
-	configs []any
-	config  Config
-	state   State
+	configs []any  `yaml:"-" mapstructure:"-"`
+	config  Config `yaml:"-" mapstructure:"-"`
```

Comment:
I thought about this a bit and you had a question, can't we just add the Application to the `allConfigs` section, below? And I realized we could just set this `config` field to `mapstructure:",squash"`, and then below just have:
```go
allConfigs := []any{a}                                 // process the core application configurations first (logging and development)
```